pydantic/dataclasses.py

```diff
@@ -178,7 +178,11 @@ def create_dataclass(cls: type[Any]) -> type[PydanticDataclass]:
 
         original_cls = cls
 
-        config_wrapper = _config.ConfigWrapper(config)
+        cls_config = getattr(cls, '__pydantic_config__', None)
+        config_dict = config
+        if config_dict is None and cls_config is not None:
+            config_dict = cls_config
+        config_wrapper = _config.ConfigWrapper(config_dict)
```

Comment:
```suggestion
        config_dict = config
        if config_dict is None:
            # if not explicitly provided, read from the type
            cls_config = getattr(cls, '__pydantic_config__', None)
            if cls_config is not None:
                config_dict = cls_config
        config_wrapper = _config.ConfigWrapper(config_dict)
```

Feel free to take or leave this change if you prefer the current implementation. I'm approving either way.